src/pdf.sandbox.js

```diff
@@ -23,87 +24,114 @@ const pdfjsBuild = PDFJSDev.eval("BUNDLE_BUILD");
 const TESTING =
   typeof PDFJSDev === "undefined" || PDFJSDev.test("!PRODUCTION || TESTING");
 
+class SandboxSupport extends SandboxSupportBase {
+  exportValueToSandbox(val) {
+    // The communication with the Quickjs sandbox is based on strings
+    // So we use JSON.stringfy to serialize
+    return this.win.JSON.stringify(val);
+  }
+
+  importValueFromSandbox(val) {
+    // ...  and JSON.parse to deserialize
+    return this.win.JSON.parse(val);
+  }
+}
+
 class Sandbox {
-  constructor(module) {
-    this._evalInSandbox = module.cwrap("evalInSandbox", null, [
-      "string",
-      "int",
-    ]);
-    this._dispatchEventName = null;
+  constructor(win, module) {
+    this.support = new SandboxSupport(win, this);
+    module.externalCall = this.support.createSandboxExternals();
+
     this._module = module;
+    this._commFun = null;
     this._alertOnError = 1;
   }
 
   create(data) {
-    const sandboxData = JSON.stringify(data);
-    const extra = [
-      "send",
-      "setTimeout",
-      "clearTimeout",
-      "setInterval",
-      "clearInterval",
-      "crackURL",
-    ];
-    const extraStr = extra.join(",");
-    let code = [
-      "exports = Object.create(null);",
-      "module = Object.create(null);",
+    const sandboxData = window.JSON.stringify(data);
```

Comment:
Use `JSON` instead of `window.JSON`. `JSON` is part of the language definition. `window.xxxx` is more appropriate for host objects such as DOM APIs.

Reply:
You had marked this as resolved but didn't replace `window.JSON` with `JSON`. Why?

Reply:
I just forgot.